packages/core/src/protocol-helpers.ts

```diff
@@ -113,16 +113,15 @@ export default class ProtocolHelpers {
         // Should interaction methods like readProperty() return an encapsulated value container with value&contenType
         // as sketched in https://github.com/w3c/wot-scripting-api/issues/201#issuecomment-573702999
         if (
-            td &&
-            propertyName &&
-            uriScheme &&
-            td.properties &&
-            td.properties[propertyName] &&
-            td.properties[propertyName].forms &&
+            propertyName != null &&
+            uriScheme != null &&
+            td?.properties != null &&
+            td.properties[propertyName] != null &&
+            td.properties[propertyName].forms != null &&
             Array.isArray(td.properties[propertyName].forms)
         ) {
             for (const form of td.properties[propertyName].forms) {
-                if (form.href && form.href.startsWith(uriScheme) && form.contentType) {
+                if (form.href && form.href.startsWith(uriScheme) && form.contentType != null) {
```

Comment:
```suggestion
                if (form.href?.startsWith(uriScheme) ?? false && form.contentType != null) {
```

Reply:
Should the following be sufficient!?

`if (form.href?.startsWith(uriScheme) && form.contentType != null) {`

Reply:
Hmm, I think in this case `form.href?.startsWith(uriScheme)` would have a type of `boolean?` (or rather `boolean | undefined`). How about `form.href?.startsWith(uriScheme) === true` as an alternative?

Reply:
What is the difference between (your proposal)
`form.href?.startsWith(uriScheme) === true`
and (mine)
`form.href?.startsWith(uriScheme)`

I am not sure why `=== true` is needed since startsWith returns always a boolean already?

Reply:
Oh, yeah, you are right, due to the `form.href` in the origin, I thought `href` could be `undefined`, but that obviously cannot be the case :) So I think we should be able to change that to `form.href.startsWith(uriScheme)` (removing the `?`).

Reply:
Okay, after double-checking locally, this actually works:

```suggestion
                if (form.href.startsWith(uriScheme) && form.contentType != null) {
```

Reply:
https://github.com/eclipse-thingweb/node-wot/pull/1096/commits/567fe1ee9446df947d9a15d1f66e9f25da43d7bd

Please speak up if you think it is wrong

Reply:
I think the `?` after the `href` could actually be removed, but the current version would also be fine :)